test/common/router/router_test.cc

```diff
@@ -147,6 +147,7 @@ TEST_F(RouterTest, UpdateServerNameFilterState) {
   NiceMock<StreamInfo::MockStreamInfo> stream_info;
   auto dummy_option = absl::make_optional<envoy::config::core::v3::UpstreamHttpProtocolOptions>();
   dummy_option.value().set_auto_sni(true);
+  dummy_option.value().set_alt_header_name(":authority");
```

Comment:
Could we also cover `auto_san_validation=true` case behavior?

Reply:
Done!